internal/auth/authenticator.go

```diff
@@ -201,38 +198,67 @@ type OAuthTokenResponse struct {
 	ExpiresIn    int    `json:"expires_in"`
 }
 
-func (d *DeviceAuthenticator) requestToken(ctx context.Context, deviceCode string, clientID string) (string, error) {
+func (d *DeviceAuthenticator) requestToken(ctx context.Context, deviceCode string, clientID string) (string, string, error) {
 	payload := strings.NewReader(fmt.Sprintf("grant_type=urn:ietf:params:oauth:grant-type:device_code&device_code=%s&client_id=%s", deviceCode, clientID))
 	req, err := d.NewFormRequest(ctx, http.MethodPost, "oauth/token", payload)
 	if err != nil {
-		return "", errors.Wrap(err, "error creating request")
+		return "", "", errors.Wrap(err, "error creating request")
 	}
 
 	res, err := d.client.Do(req)
 	if err != nil {
-		return "", errors.Wrap(err, "error performing http request")
+		return "", "", errors.Wrap(err, "error performing http request")
 	}
 
 	defer res.Body.Close()
 
 	isRetryable, err := checkErrorResponse(res)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	// Bail early so the token fetching is retried.
 	if isRetryable {
-		return "", nil
+		return "", "", nil
 	}
 
 	tokenRes := &OAuthTokenResponse{}
 
 	err = json.NewDecoder(res.Body).Decode(tokenRes)
 	if err != nil {
-		return "", errors.Wrap(err, "error decoding token response")
+		return "", "", errors.Wrap(err, "error decoding token response")
+	}
+
+	return tokenRes.AccessToken, tokenRes.RefreshToken, nil
+}
+
+// RefreshToken for an access token
```

Comment:
Note the meaning of each of the returned strings in the doc comment. Named return values are also doable but I tend to prefer better doc comments instead.